kasa/smartprotocol.py

```diff
@@ -157,15 +147,50 @@ async def _execute_query(self, request: Union[str, Dict], retry_count: int) -> D
             self._host,
             _LOGGER.isEnabledFor(logging.DEBUG) and pf(smart_request),
         )
-        response_data = await self._transport.send(smart_request)
+        resp_dict = await self._transport.send(smart_request)
 
         _LOGGER.debug(
             "%s << %s",
             self._host,
-            _LOGGER.isEnabledFor(logging.DEBUG) and pf(response_data),
+            _LOGGER.isEnabledFor(logging.DEBUG) and pf(resp_dict),
         )
 
-        return response_data
+        self._handle_response_error_code(resp_dict)
+
+        if result := resp_dict.get("result"):
+            if is_multi:
+                if (responses := result.get("responses")) is None:
+                    raise SmartDeviceException(
+                        "Unexpected response to multipleRequest "
+                        + f"method for {self._host}: {result}"
+                    )
+                multi_result = {}
+                for response in responses:
+                    self._handle_response_error_code(response)
+                    result = response.get("result", {})
+                    multi_result[response["method"]] = result
+                return multi_result
+
+            return {smart_method: result}
+        return {smart_method: None}
+
+    def _handle_response_error_code(self, resp_dict: dict):
+        if (
+            error_code := SmartErrorCode(resp_dict.get("error_code"))  # type: ignore[arg-type]
+        ) != SmartErrorCode.SUCCESS:
```

Comment:
I know this was already there like this, but this would be much simpler if we return only on successes (avoids nesting):
```
error_code = SmartErrorCode(resp_dict.get("error_code")
if not error_code:
    return

<handle error cases here>
```

Reply:
Done, also for the same method in the `AesTransport` class.